testing/mock/v2/ibc_module.go

```diff
@@ -0,0 +1,60 @@
+package mock
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	channeltypesv2 "github.com/cosmos/ibc-go/v9/modules/core/04-channel/v2/types"
+	"github.com/cosmos/ibc-go/v9/modules/core/api"
+)
+
+var _ api.IBCModule = (*IBCModule)(nil)
+
+const (
+	// ModuleNameA is a name that can be used for the first mock application.
+	ModuleNameA = ModuleName + "A"
+	// ModuleNameB is a name that can be used for the second mock application.
+	ModuleNameB = ModuleName + "B"
+)
+
+// IBCModule is a mock implementation of the IBCModule interface.
+// which delegates calls to the underlying IBCApp.
+type IBCModule struct {
+	IBCApp *IBCApp
+}
+
+// NewIBCModule creates a new IBCModule.
+func NewIBCModule() IBCModule {
+	return IBCModule{
+		IBCApp: &IBCApp{},
+	}
+}
```

Comment:
modified signature here to not include the IBCApp as an argument, think it's cleaner this way and don't think it would ever be required to pass a custom IBCApp at module creation time.